Mathlib/Topology/Algebra/Order/Archimedean.lean

```diff
@@ -66,4 +67,24 @@ theorem dense_or_cyclic (S : AddSubgroup G) : Dense (S : Set G) ∨ ∃ a : G, S
   rcases h with ⟨ε, ε0, hε⟩
   exact cyclic_of_isolated_zero ε0 (disjoint_left.2 hε)
 
+variable [Nontrivial G] [DenselyOrdered G]
+
+/-- In a nontrivial densely linear ordered archimedean topological group,
+a subgroup is either dense or is cyclic, but not both.
+
+For a non-exclusive "or" version, see `AddSubgroup.dense_or_cyclic` above. -/
+theorem dense_xor'_cyclic (s : AddSubgroup G) :
+    Xor' (Dense (s : Set G)) (∃ a, s = .zmultiples a) := by
```

Comment:
Should [LinearOrderedCommGroup.discrete_iff_not_denselyOrdered](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/ArchimedeanDensely.html#LinearOrderedCommGroup.discrete_iff_not_denselyOrdered) be also stated with Xor'?